### ISSUE TITLE
Properly shutdown SSH multiplexing

### ISSUE THREAD

**Issue by jonesetc** [github/backup-utils#349]
github-backup-utils: 2.11.1
Bug introduced: #321 

Unless `GHE_DISABLE_SSH_MUX` is set (slowing everything down to old speeds), the master SSH connection is left without clean up. This is removed after the 10 minute timeout is reached, but should be cleaned up manually.

I am running this backup from an EC2 instance with the backup data directory mounted as an EBS volume. This long living master connection makes it so that I can't safely unmount the EBS volume until that 10 minute timeout as the ssh as is it launched from inside of the data directory.

The way to clean up this master connection is `ssh host -O exit`, which passes the exit command specifically to the master connection.

```
# See that we can't unmount
$ sudo umount /var/run/backups-github/data
umount: /var/run/backups-github/data: target is busy
        (In some cases useful info about processes that
         use the device is found by lsof(8) or fuser(1).)

# See that there are two processes being run from the mountpoint, stopping the unmount
$ sudo fuser -c /var/run/backups-github/data
/var/run/backups-github/data:           31856c 31889c

# Look up both processes and see that they are the master SSH processes
$ sudo ps aux | grep 31856
worker  31856  0.0  0.0  46988   704 ?        Ss   18:51   0:00 ssh -f -p 22 -o ControlMaster=auto -o ControlPath="/tmp/.ghe-sshmux-bd4bf3de" -o ControlPersist=10m -o ServerAliveInterval=10 -l admin -o StrictHostKeyChecking=no -o BatchMode=no -o PasswordAuthentication=no -o ConnectTimeout=5 -o ConnectionAttempts=1 -o BatchMode=yes github.internal.com -- /bin/true
$ sudo ps aux | grep 31889
worker  31889  0.8  0.0  47196  2868 ?        Ss   18:51   0:03 ssh -f -p 122 -o ControlMaster=auto -o ControlPath="/tmp/.ghe-sshmux-59580432" -o ControlPersist=10m -o ServerAliveInterval=10 -l admin -o StrictHostKeyChecking=no -o BatchMode=no -o PasswordAuthentication=no -o ConnectTimeout=5 -o ConnectionAttempts=1 -o BatchMode=yes github.internal.com -- /bin/true
```

It appears that two master SSH processes are created as well, but that seems like another issue.

**Comment by lildude** [comment]
I believe I fixed this issue the other day when fixing this for https://github.com/github/backup-utils/issues/345 but haven't made a new release yet. Can you please try the master branch and let me know if you're still seeing the issue.

**Comment by jonesetc** [comment]
That does look like it would solve this issue for me, as the master processes would be run in `$TEMPDIR` instead of the data directory. I'll give it a test tomorrow, thanks.

**Comment by lildude** [comment]
I released version [2.11.2](https://github.com/github/backup-utils/releases/tag/v2.11.2) today which includes the fix for #345 and should address this issue too.

Closing, but if this isn't completely resolved, please feel free to re-open.
